test/app-tap/datetime.test.lua

```diff
@@ -1830,6 +1830,565 @@ test:test("Parse strptime format", function(test)
+            fmt = '%j',
+        },
+        {
+            buf = '31-12-2024 366',
+            fmt = '%d-%m-%Y %j',
+        },
+        -- %k
+        {
+            buf = ' 0',
+            fmt = '%k',
+        },
+        {
+            buf = '23',
+            fmt = '%k',
+        },
+        -- %l
+        {
+            buf = ' 1',
+            fmt = '%l',
+        },
+        {
+            buf = '12',
+            fmt = '%l',
+        },
+        -- %m
+        {
+            buf = '01',
+            fmt = '%m',
+        },
+        {
+            buf = '12',
+            fmt = '%m',
+        },
+        -- %M
+        {
+            buf = '00',
+            fmt = '%M',
+        },
+        {
+            buf = '59',
+            fmt = '%M',
+        },
+        -- %n
+        {
+            buf = '\n',
+            fmt = '%n',
+        },
+        -- %O, modifier, skipped.
+        -- %p
+        {
+            buf = 'PM',
+            fmt = '%p',
+        },
+        {
+            buf = 'AM',
+            fmt = '%p',
+        },
+        -- %r, same as "%I:%M:%S %p".
+        {
+            buf = '12:32:01 AM',
+            fmt = '%r',
+        },
+        {
+            buf = '12:32:01 PM',
+            fmt = '%r',
+        },
+        -- %R, same as "%H:%M".
+        {
+            buf = '18:20',
+            fmt = '%R',
+        },
+        -- %s
+        {
+            buf = '26-08-2024 1724630400',
+            fmt = '%d-%m-%Y %s',
+        },
+        -- %S
+        {
+            buf = '00',
+            fmt = '%S',
+        },
+        {
+            buf = '59',
+            fmt = '%S',
+        },
+        -- %t
+        {
+            buf = '26-08-2024 \t',
+            fmt = '%d-%m-%Y %t',
+        },
+        -- %T, same as "%H:%M:%S".
+        {
+            buf = '00:00:00',
+            fmt = '%T',
+        },
+        {
+            buf = '23:59:59',
+            fmt = '%T',
+        },
+        -- %u
+        -- Works for 4 ('Thursday') only when date is not
+        -- complete.
+        {
+            buf = '4',
+            fmt = '%u',
+        },
+        {
+            buf = '30-08-2024 5',
+            fmt = '%d-%m-%Y %u',
+        },
+        {
+            buf = '26-08-2024 1',
+            fmt = '%d-%m-%Y %u',
+        },
+        -- %U
+        {
+            buf = '01-01-2024 00',
+            fmt = '%d-%m-%Y %U',
+        },
+        {
+            buf = '31-12-2024 52', -- "00 to 53"?
+            fmt = '%d-%m-%Y %U',
+        },
+        -- %v, same as "%e-%b-%".
+        {
+            buf = ' 1-Jan-1970',
+            fmt = '%v',
+        },
+        -- %V, unsupported by strptime(3).
+        -- %w
+        -- Works for 4 (Thursday) only when date is not complete.
+        {
+            buf = '4',
+            fmt = '%w',
+        },
+        {
+            buf = '26-08-2024 1',
+            fmt = '%d-%m-%Y %w',
+        },
+        -- %W
+        {
+            buf = '26-08-2024 35',
```

Comment:
May be it is worth adding the corner cases here too, as for `%U` format.

Reply:
Seems `%W` has no full support [^1], any value is accepted and `week number` does not expose datetime object:

[^1]: https://github.com/tarantool/tarantool/issues/10575

```lua
tarantool> datetime.parse('01-01-2024 34', {format ='%d-%m-%Y %W'})
---
- 2024-01-01T00:00:00Z
- 13
...

tarantool> datetime.parse('01-01-2024 32', {format ='%d-%m-%Y %W'})
---
- 2024-01-01T00:00:00Z
- 13
...

tarantool> datetime.parse('01-01-2024 31', {format ='%d-%m-%Y %W'})
---
- 2024-01-01T00:00:00Z
- 13
...

tarantool> datetime.parse('01-01-2024 29', {format ='%d-%m-%Y %W'})
---
- 2024-01-01T00:00:00Z
- 13
```

So there is no much sense in thorough testing here.
Anyway, I'll add edge cases according to [^2].

[^2]: https://www.epochconverter.com/weeks/2024



```diff
diff --git a/test/app-tap/datetime.test.lua b/test/app-tap/datetime.test.lua
index 2582bbb4c3..b750faba45 100755
--- a/test/app-tap/datetime.test.lua
+++ b/test/app-tap/datetime.test.lua
@@ -2434,10 +2434,43 @@ test:test("Parse with a custom format and format back to string", function(test)
         -- %W is replaced by the week number of the year (Monday
         -- as the first day of the week) as a decimal number
         -- (00-53).
+        -- Note, week number is not supported,
+        -- see tarantool/tarantool#10575.
+        -- First day of the first week of the year.
         {
-            buf = '26-08-2024 35',
+            buf = '01-01-2024 01',
             fmt = '%d-%m-%Y %W',
-            dt = date.new({year = 2024, month = 08, day = 26}),
+            dt = date.new({year = 2024, month = 01, day = 01}),
+        },
+        -- Last day of the first week of the year.
+        {
+            buf = '07-01-2024 01',
+            fmt = '%d-%m-%Y %W',
+            dt = date.new({year = 2024, month = 01, day = 07}),
+        },
+        -- First day of the second week of the year.
+        {
+            buf = '08-01-2024 02',
+            fmt = '%d-%m-%Y %W',
+            dt = date.new({year = 2024, month = 01, day = 08}),
+        },
+        -- First day of the last week of the year.
+        {
+            buf = '23-12-2024 52',
+            fmt = '%d-%m-%Y %W',
+            dt = date.new({year = 2024, month = 12, day = 23}),
+        },
+        -- Last day of the last week of the year.
+        {
+            buf = '29-12-2024 52',
+            fmt = '%d-%m-%Y %W',
+            dt = date.new({year = 2024, month = 12, day = 29}),
+        },
+        -- Last day of the year.
+        {
+            buf = '31-12-2024 53',
+            fmt = '%d-%m-%Y %W',
+            dt = date.new({year = 2024, month = 12, day = 31}),
         },
         -- %x is replaced by national representation of the date.
         {
```

Reply:
Nice, thanks!